phpcs.xml

```diff
@@ -2,6 +2,17 @@
 <ruleset name="Zend Framework Coding Standard">
     <rule ref="./vendor/zendframework/zend-coding-standard/ruleset.xml"/>
 
+    <!--
+        Don't make it bail at the file level deprecation of AbstactInjection (Intented side effect).
+
+        The inline ignores phpcs:disable or @codingStandardsIgnoreStart (as documented on
+        https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-filework)
+        do NOT work, phpcs still errors. This is the only way to fix it.
+    -->
+    <rule ref="PSR1.Files.SideEffects.FoundWithSymbols">
+        <exclude-pattern>src/Resolver/AbstractInjection.php</exclude-pattern>
```

Comment:
This is the only option that worked. Better ideas. Inline annotations had no effect (tested).

Reply:
Which annotations were you trying to use?

If you're using the current 1.0.0 release branch of zend-coding-standard, you'll need to use:

```php
// @codingStandardIgnoreStart
some code
// @codingStandardIgnoreEnd
```

If you are using a newer version of phpcs (v3.1+, IIRC), these become simpler:

```php
// phpcs:disable
some code
// phpcs:enable
```

My guess is you were trying the latter, when the tool is looking for the former.

Reply:
zend-coding-standard 1.0.0 is in use here, but using `@codingStandardIgnoreStart` and `@codingStandardIgnoreEnd` still errors on my machine.

I pushed a commit to test it on travis. If this errors as well, this may be a bug in phpcs.

Reply:
And here we go.

This Commit: https://github.com/zendframework/zend-di/pull/38/commits/7e3d315fee2aca538a52ae89600b4fe6be68a752 will now error:
https://travis-ci.org/zendframework/zend-di/jobs/427141904#L657

Possibly a phpcs bug. I'll revert this commit.

Reply:
There's another option here, since you're trying to ignore the file entirely: one of either `phpcs:ignoreFile` or `@codingStandardsIgnoreFile` (former is for phpcs 3.1 and up, latter is for previous versions). These should be put in the first line(s) of the file.

Reply:
Were you ever able to resolve this?

Reply:
Unfortunately, no. Not with the phpcs version used by the current zend-coding-standard. This is the only way without sacrificing other style checks. 

Since the wip CS 2 updates phpcs as well, this issue with the not working annotations may be fixed and this can hoefully be ditched. Until then I'd reason to do it this way.